task01/src/Vector.cpp

```diff
@@ -0,0 +1,318 @@
+        delete[] m_array;
+        m_size = other.m_size;
+        m_capacity = other.m_capacity;
+        m_array = other.m_array;
+        other.m_size = 0;
+        other.m_capacity = 0;
+        other.m_array = nullptr;
+    }
+    return *this;
+}
+
+//--------------------------------------------------------------------------------
+
+void Vector::push_back(const int& value) {
+    // Check if we need to resize the array
+    if (m_size >= m_capacity) {
+        // Double the capacity to reduce the number of reallocations
+        std::size_t new_capacity = m_capacity * 2;
+        int* new_array = new int[new_capacity];
+
+        // Copy the existing elements to the new array
+        for (std::size_t i = 0; i < m_size; i++) {
+            new_array[i] = m_array[i];
+        }
+
+        // Delete the old array and update the pointer and capacity
+        delete[] m_array;
+        m_array = new_array;
+        m_capacity = new_capacity;
+    }
+
+    // Add the new element to the end of the array and update the size
+    m_array[m_size] = value;
+    m_size++;
+}
+
+//--------------------------------------------------------------------------------
+
+void Vector::resize(std::size_t count) {
+	resize(count, 0);
+	
+}
+
+//--------------------------------------------------------------------------------
+
+void Vector::resize(std::size_t count, const int& value) {
+    if (count <= m_capacity) {
+        if (count < m_size) {
+            // Truncate the vector by updating its size
+            m_size = count;
+        } else if (count > m_size) {
+            // Append the specified value to the vector for the remaining elements
+            for (std::size_t i = m_size; i < count; ++i) {
+                m_array[i] = value;
+            }
+            m_size = count;
+        }
+    } else {
+        // Allocate a new buffer with appropriate size
+        std::size_t new_capacity = m_capacity * 2;
+        while (new_capacity < count) {
+            new_capacity *= 2;
+        }
+
+        // Create new array and copy elements from old array to new array
+        int* new_array = new int[new_capacity];
+        for (std::size_t i = 0; i < m_size; ++i) {
+            new_array[i] = m_array[i];
+        }
+        // Append the specified value to the vector for the remaining elements
+        for (std::size_t i = m_size; i < count; ++i) {
+            new_array[i] = value;
+        }
+
+        // Deallocate the old buffer and update the vector's fields
+        delete[] m_array;
+        m_array = new_array;
+        m_capacity = new_capacity;
+        m_size = count;
+    }
+}
+
+//--------------------------------------------------------------------------------
+/* data() function whitch returns a pointer of first cell*/
+int* Vector::data() const {
+	if (m_size != 0) {
+		return &m_array[0];
+	}
+}
+
+//--------------------------------------------------------------------------------
+void Vector::clear() {
+	m_size = 0;
+	m_array = 0;
```

Comment:
Don't set a `int *` directly with `0`! This is very bad practice.  You can use `nullptr` if you need.
Also, in this case you're resetting your pointer and thus creating a memory leak. You need to `delete[] ptr` and pointer before resetting it